src/k8s/pkg/k8sd/database/feature_status_test.go

```diff
@@ -0,0 +1,107 @@
+package database_test
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/canonical/k8s/pkg/k8sd/database"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+)
+
+func TestFeatureStatus(t *testing.T) {
+	WithDB(t, func(ctx context.Context, db DB) {
+		_ = db.Transaction(ctx, func(ctx context.Context, tx *sql.Tx) error {
+			g := NewWithT(t)
+			t0, _ := time.Parse(time.RFC3339, time.Now().Format(time.RFC3339))
+			// initial get should return nothing
```

Comment:
instead of comments, let's make these subtests with `t.Run(...)`.

See the existing tests for ideas